dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preprocess/AssignedUserPreProcessor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.preprocess;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.hisp.dhis.tracker.bundle.TrackerBundle;
+import org.hisp.dhis.tracker.domain.Event;
+import org.hisp.dhis.tracker.domain.UserInfo;
+import org.hisp.dhis.user.User;
+import org.springframework.stereotype.Component;
+
+/**
+ * This preprocessor is responsible to fill the assigned user with missing
+ * information.
+ *
+ * @author Enrico Colasante
+ */
+@Component
+public class AssignedUserPreProcessor
+    implements BundlePreProcessor
+{
+    @Override
+    public void process( TrackerBundle bundle )
+    {
+        for ( Event event : bundle.getEvents() )
+        {
+            UserInfo assignedUser = event.getAssignedUser();
+
+            if ( Objects.nonNull( assignedUser ) )
+            {
+
+                if ( Objects.isNull( assignedUser.getUid() ) && Objects.nonNull( assignedUser.getUsername() ) )
```

Comment:
since you are using optionals, could we just do 

```suggestion
                if ( Objects.isNull( assignedUser.getUid() ) )
```

same for the next case. And make both just independent if statements.

Reply:
Done